controller/replicator.go

```diff
@@ -312,6 +312,95 @@ func (r *replicator) Snapshot(name string, userCreated bool, created string) err
 	return nil
 }
 
+func (r *replicator) GetLastSnapshot() (string, error) {
+	retErrorLock := sync.Mutex{}
+	retError := &BackendError{
```

Comment:
why a struct over map? can't we just create a map `retErrors map[string][]string` similar to replicaChains

Reply:
This has to be returned as an error. That  is why it has been been in a struct. It is a common practice being followed in other functions also.